src/main/java/org/logstash/beats/MessageListener.java

```diff
@@ -0,0 +1,29 @@
+package org.logstash.beats;
+
+import io.netty.channel.ChannelHandlerContext;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+
+// This need to be implemented in Ruby
+public class MessageListener implements IMessageListener {
```

Comment:
JRuby does fine with interface, you just need to include it?
I dont see any specific issue here.